x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_isolation_status.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import { useEffect, useState } from 'react';
+import { Maybe } from '../../../../../../observability/common/typings';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { getHostMetadata } from './api';
+import { ISOLATION_STATUS_FAILURE } from './translations';
+
+interface HostIsolationStatusResponse {
+  loading: boolean;
+  isIsolated: Maybe<boolean>;
+}
+
+/*
+ * Retrieves the current isolation status of a host */
+export const useHostIsolationStatus = ({
+  agentId,
+}: {
+  agentId: string;
+}): HostIsolationStatusResponse => {
+  const [isIsolated, setIsIsolated] = useState<Maybe<boolean>>();
+  const [loading, setLoading] = useState(false);
+
+  const { addError } = useAppToasts();
+
+  useEffect(() => {
+    // isMounted tracks if a component is mounted before changing state
+    let isMounted = true;
+    setLoading(true);
+    const fetchData = async () => {
+      try {
+        const metadataResponse = await getHostMetadata({ agentId });
+        if (isMounted) {
+          setIsIsolated(Boolean(metadataResponse.metadata.Endpoint.configuration.isolation));
```

Comment:
I think we have now agreed that this should be `metadata.Endpoint.state.isolation` instead

Reply:
Also FYI - because I'm doing the same thing in Redux middleware, I create a separate function to do this check using the `HostMetadata` as input - I will move that to the common folder of endpoint isolation and will then re-factor this here to use it as well. Like that we remain in sync as to how this is determined.

Reply:
+1

although this does create a situation in which someone can be stuck in an unwanted state:

1. Click `isolate` on a host
2. Endpoint receives the action, writes the intent to policy config
3. _applying_ the policy fails (So at this point, `Endpoint.configuration.isolation: true` and `Endpoint.state.isolation: false`).
4. The button we present is `isolate`. Which is akin to a retry.
5. If it continues to fail (maybe something persistent is wrong, like a kernel extension is blocked), then the configuration will remain as `isolate: true` until the problem is resolved. Let's say some weeks later, IT is troubleshooting something else, notices the blocked kext, and loads it. Boom, isolated. The action will not have expired because the endpoint applied it, it will be there until the reverse action is sent.

There is no way to undo when you're in the incongruent state

Reply:
gotcha